go.mod

```diff
@@ -553,7 +553,7 @@ replace (
 	github.com/alecthomas/kingpin/v2 => github.com/gravitational/kingpin/v2 v2.1.11-0.20230515143221-4ec6b70ecd33
 	github.com/coreos/go-oidc => github.com/gravitational/go-oidc v0.1.1
 	github.com/crewjam/saml => github.com/gravitational/saml v0.4.15-teleport.1
-	github.com/datastax/go-cassandra-native-protocol => github.com/gravitational/go-cassandra-native-protocol v0.0.0-20221005103706-b9e66c056e90
+	github.com/datastax/go-cassandra-native-protocol => github.com/gravitational/go-cassandra-native-protocol v0.0.0-teleport.1
```

Comment:
@smallinsky I am planning to make an upstream PR to either expose the listener or have a function to return the listener address.

Do you recall why the other changes are needed?
- https://github.com/gravitational/go-cassandra-native-protocol/commit/b9e66c056e909bc9c96a25f159bf16269cac1b44
- https://github.com/gravitational/go-cassandra-native-protocol/commit/62313b8297faeccea6af15f334397c1a24588df2

Reply:
I can't recall full context but I think that there were some race condition in our db engine tests after exposing the listener address. 

Maybe this has been fixed but please ensure that after alignment the UT with updated dependency passes with race flag.